uarray/backend.py

```diff
@@ -134,33 +179,188 @@ def inner(dispatcher: ArgumentExtractorType) -> MultiMethod:
 InstanceStubType = Any
 ConvertorType = Callable[["DispatchableInstance"], Any]
 MethodLookupType = Dict[Optional[MultiMethod], ImplementationType]
-TypeLookupType = Dict[Type["DispatchableType"], ConvertorType]
+TypeLookupType = Dict[Type["DispatchableInstance"], ConvertorType]
 InstanceLookupType = Dict["DispatchableInstance", InstanceStubType]
 
 
 class Backend(metaclass=ABCMeta):
+    """
+    An abstract base class for all backend types.
+
+    See Also
+    --------
+    TypeCheckBackend: A specific kind of Backend.
```

Comment:
formatting: needs a space before `:` 

(is the case in more places, I'm not 100% sure if numpydoc/napoleon render it correctly otherwise)